### PR TITLE
Create menu component and profile button

### LINKED ISSUE #237: Create menu component and profile button
### Description
Create general menu

### Requirements

- general menu must have slot for any child as source
- must be responsive
- must be dark-mode ready

### Deliverables and requirements

- `MenuSlim` - simple menu with links
- `ProfileButton` - profile button with image

### Reference

image

### PR DESCRIPTION
**Pull request by sulton-max**
Closes #237

### CODE REVIEW

File: src/Training.AirBnb.Clone.Backend/AirBnB.Api/appsettings.Development.json [hunk 1]
```diff
@@ -36,7 +36,12 @@
   },
 
   "ConnectionStrings": {
+<<<<<<< Updated upstream
     "DbConnectionString": "Host=localhost;Port=5432;Database=Training.AirBnb.Clone;Username=postgres;Password=postgres",
+=======
+    "DefaultConnection": "Host=localhost;Port=5432;Database=Training.AirBnb.Clone;Username=postgres;Password=java2001",
+    "NotificationConnection": "Host=localhost;Port=5432;Database=Training.AirBnB.Notifications;Username=postgres;Password=java2001",
+>>>>>>> Stashed changes
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Api/appsettings.Development.json:44]: Remove git merging labels

File: src/Training.AirBnb.Clone.Backend/AirBnB.Persistence/Migrations/AppDbContextModelSnapshot.cs
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Persistence/Migrations/AppDbContextModelSnapshot.cs:1]: Remove git merging labels

File: src/Training.AirBnb.Clone.Backend/AirBnB.Api/appsettings.Development.json (hunk 1 above)
Comment (sulton-max) [reply src/Training.AirBnb.Clone.Backend/AirBnB.Api/appsettings.Development.json:44]: @Ilxom-22 we don't need multiple connection strings right ?

File: src/Training.AirBnb.Clone.Backend/AirBnB.Persistence/Migrations/20240107124530_RoleMigrations.cs [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+                name: "UserSettings",
+                schema: "notification",
+                columns: table => new
+                {
+                    Id = table.Column<Guid>(type: "uuid", nullable: false),
+                    PreferredTheme = table.Column<int>(type: "integer", nullable: false),
+                    PreferredNotificationType = table.Column<int>(type: "integer", nullable: false),
+                    UserId = table.Column<Guid>(type: "uuid", nullable: false),
+                    IsDeleted = table.Column<bool>(type: "boolean", nullable: false),
+                    DeletedTime = table.Column<DateTimeOffset>(type: "timestamp with time zone", nullable: true)
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Persistence/Migrations/20240107124530_RoleMigrations.cs:46]: Why we have these tables in migrations ?

File: src/Training.AirBnb.Clone.Frontend/src/common/views/HomeView.vue [hunk 3]
```diff
@@ -7,10 +7,10 @@
   
   <!--Listings-->
   <listings-grid class="mt-24"/>
-  
-  
-  <button class="mt-80" @click="appThemeService.toggleDarkMode()">Dark mode</button>
+
 </div>
+=======
+>>>>>>> Stashed changes
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Frontend/src/common/views/HomeView.vue:13]: Remove git labels

File: src/Training.AirBnb.Clone.Frontend/src/Modules/profile/components/ProfileMenuComponent.vue [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    <button class="text-left theme-text-primary font-sans font-light p-3 w-full theme-hover-shadow">
+      <a href="http://localhost:5173/">Help Center</a>
+    </button>
+
+    <div class="h-[1px] flex top-[150px] w-full border-b"/>
+
+    <div class="h-10 w-full flex items-center max-w-full max-h-full mb-2 mt-1">
+      <span class="mr-[75px] text-left theme-text-primary font-sans font-light pl-3">Dark mode</span>
+      <darkmode v-model="toggleState"/>
+    </div>
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Frontend/src/Modules/profile/components/ProfileMenuComponent.vue:30]: Create a model for menu and render buttons with v-for

File: src/Training.AirBnb.Clone.Backend/AirBnB.Api/appsettings.Development.json [hunk 5]
```diff
@@ -37,6 +37,8 @@
 
   "ConnectionStrings": {
     "DbConnectionString": "Host=localhost;Port=5432;Database=Training.AirBnb.Clone;Username=postgres;Password=postgres",
+    "DefaultConnection": "Host=localhost;Port=5432;Database=Training.AirBnb.Clone;Username=postgres;Password=postgres",
+    "NotificationConnection": "Host=localhost;Port=5432;Database=Training.AirBnB.Notifications;Username=postgres;Password=postgres",
```
Comment (Ilxom-22) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Api/appsettings.Development.json:41]: DefaultConnection bilan NotificationConnection string larni endi keragi yo'q. DbContext bitta bo'ladi.

File: src/Training.AirBnb.Clone.Backend/AirBnB.Persistence/Migrations/20240107124530_RoleMigrations.Designer.cs [hunk 6]
```diff
@@ -0,0 +1,407 @@
...
+                {
+                    b.Property<Guid>("Id")
+                        .ValueGeneratedOnAdd()
+                        .HasColumnType("uuid");
+
+                    b.Property<DateOnly>("BuiltDate")
+                        .HasColumnType("date");
+
+                    b.Property<DateTimeOffset?>("DeletedTime")
+                        .HasColumnType("timestamp with time zone");
```
Comment (Ilxom-22) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Persistence/Migrations/20240107124530_RoleMigrations.Designer.cs:39]: Role Migration bu PR da bo'lishi kerak emas edi

**Review by Ilxom-22 — CHANGES_REQUESTED** [review]

File: src/Training.AirBnb.Clone.Backend/AirBnB.Api/appsettings.Development.json (hunk 1 above)
Comment (Abdurasulovich) [reply src/Training.AirBnb.Clone.Backend/AirBnB.Api/appsettings.Development.json]: Which connections we don't need?

**Pull request closed without merging**